### ISSUE TITLE
Build error: 'const' declarations are only available when targeting ECMAScript 6 and higher.

### ISSUE THREAD

**Issue by sstorie** [Justin-Credible/Ionic-TypeScript-MDHA-Starter#36]
I just grabbed the latest code for this repo, and tried to just build it without any changes and am getting the error in the subject for the Constants.ts file. I am searching around to see where the ES version is specified, but am a bit lost here. I assume this is an issue with my machine somehow, but if anyone has any thoughts on why the const keyword isn't just working I'd appreciate the help.

Btw, thank you so much for putting this repo out there. Coming from 10 years of experience in the .NET world (both desktop and server apps) I've never felt so lost as I am now trying to get started with Angular, Ionic, Typescript & Visual Studio. I understand Angular, but trying to integrate Typescript and Visual Studio has been very frustrating. Once you add something like Cordova/Ionic it gets even more maddening! Working examples like this are a god send :)

**Comment by sstorie** [comment]
Ok, to make absolutely sure I had everything i should i re-installed the Typescript 1.5 beta and re-opened the solution once that was in place, and now I don't get those const errors while building. Instead I get this error:

```
Severity    Code    Description Project File    Line
Error       The "RunMdaInstall" task was not given a value for the required parameter "NpmInstallDir".  JustinCredible.SampleApp    C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\ApacheCordovaTools\vs-mda-targets\Microsoft.MDA.FileMirroring.targets   203

```

From what I can tell node.js is installed just fine (I can build js-based projects outside of VS2015 without issue), and even from the VS command prompt npm is available:

```
C:\Program Files (x86)\Microsoft Visual Studio 14.0>npm config list
; cli configs
user-agent = "npm/2.13.0 node/v0.12.7 win32 x64"

; builtin config undefined
prefix = "C:\\Users\\samstor\\AppData\\Roaming\\npm"

; node bin location = C:\Program Files\nodejs\node.exe
; cwd = C:\Program Files (x86)\Microsoft Visual Studio 14.0
; HOME = C:\Users\samstor
; 'npm config ls -l' to show all defaults.
```

Any thoughts on this particular error?

**Comment by Justin-Credible** [comment]
Hi @sstorie, I'm glad you've found this project useful.

When I added the usage of the `const` keyword the main thing I had to do was manually install the 1.5 beta of TypeScript and then update the jsproj file setting `TypeScriptToolsVersion` to 1.5.

I didn't run into the error that you did. The only thing I can think of to try differently is your node version. I'm pretty sure I'm using 0.10.x and not 0.12.x.

**Comment by sstorie** [comment]
Ok I'll keep poking around to see if I can figure this out.

**Comment by sstorie** [comment]
Ok, my second issue was caused by me. In a previous attempt to make sure I was using the latest version of npm I had removed the `npm` and `npm.cmd` files from the nodejs install directory. This was making my machine use the npm version installed in %APPDATA%, which I could update easily outside of node.js itself. However, this seems to break visual studio's ability to compile a Cordova application. Once I restored those two npm files back this started working as expected.

I can now build your code without any problems. Thanks again for replying.
